messaging/index.js

```diff
@@ -45,20 +50,22 @@ export class MessagingContext {
 }
 
 /**
- * @typedef {WebkitMessagingConfig | WindowsMessagingConfig | AndroidMessagingConfig | TestTransportConfig} MessagingConfig
- */
-
-/**
- *
+ * The consumer interface
  */
 export class Messaging {
     /**
      * @param {MessagingContext} messagingContext
-     * @param {MessagingConfig} config
+     * @param {MessagingTransport} transport
```

Comment:
the significant change here, is that we directly take a transport now, preventing the core library needing to know about them all ahead of time